instance/models/openedx_instance.py

```diff
@@ -117,12 +117,19 @@ def get_load_balancer_configuration(self, triggered_by_instance=False):
         for appserver in active_appservers:
             server_name = "appserver-{}".format(appserver.pk)
             ip_address = appserver.server.public_ip
+            if not ip_address:
+                # If the ip_address is not there, try calling update_status(). If the IP is still not set
+                # then fail the load balancer reconfiguration process.
+                appserver.server.update_status()
+                ip_address = appserver.server.public_ip
             if ip_address:
                 appserver_vars.append(dict(ip_address=ip_address, name=server_name))
             else:
                 appserver.logger.error(
                     "Active appserver does not have a public IP address. This should not happen."
+                    "Deconfiguring the load balancer backend."
                 )
+                return [], []
```

Comment:
@UmanShahzad , this is pretty similar to what's happening on line 139. Raising an Exception would mess with the entire call stack.